lib/src/migrators/module.dart

```diff
@@ -738,110 +739,173 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     super.visitFunctionRule(node);
   }
 
-  /// Migrates an `@import` rule to a `@use` rule after migrating the imported
-  /// file.
+  /// Visits an `@import` rule, migrating it to one or more `@use`, `@forward`,
+  /// `meta.load-css`, or static `@import` rules
   @override
   void visitImportRule(ImportRule node) {
     var imports =
         partitionOnType<Import, StaticImport, DynamicImport>(node.imports);
     var staticImports = imports.item1;
-    var dynamicImports = imports.item2.where((import) {
-      var url = importCache
-          .canonicalize(Uri.parse(import.url),
-              baseImporter: importer, forImport: true)
-          ?.item2;
-      return !references.orphanImportOnlyFiles.contains(url);
-    });
+    var dynamicImports = imports.item2;
+    if (dynamicImports.isEmpty) {
+      _useAllowed = false;
+      return;
+    }
+    String rulesText;
+
+    var migratedRules = <String>[];
+
+    var indent = ' ' * node.span.start.column;
 
-    var start = node.span.start;
-    var first = true;
     for (var import in dynamicImports) {
-      if (first) {
-        first = false;
-      } else {
-        addPatch(Patch.insert(start, "$_semicolonIfNotIndented\n"));
+      var ruleUrl = import.url;
+      var canonicalImport = importCache
+          .canonicalize(Uri.parse(ruleUrl),
+              baseImporter: importer, forImport: true)
+          ?.item2;
+      if (references.orphanImportOnlyFiles.containsKey(canonicalImport)) {
+        ruleUrl =
+            references.orphanImportOnlyFiles[canonicalImport]?.url.toString();
       }
 
-      _migrateImport(import, start);
+      if (ruleUrl != null) {
+        if (_useAllowed) {
+          migratedRules.addAll(_migrateImportToRules(ruleUrl, import.span));
+        } else {
+          migratedRules.add(_migrateImportToLoadCss(ruleUrl, import.span)
+              .replaceAll('\n', '\n$indent'));
+        }
+      }
     }
 
-    var spanWithNewline = extendForward(
-            extendThroughWhitespace(node.span), '$_semicolonIfNotIndented\n') ??
-        node.span;
+    rulesText = migratedRules.join('$_semicolonIfNotIndented\n$indent');
 
-    if (staticImports.isNotEmpty) {
-      if (dynamicImports.isNotEmpty) {
-        addPatch(Patch.insert(start, "$_semicolonIfNotIndented\n"));
-      }
+    addPatch(Patch(node.span, rulesText));
+    if (_useAllowed) {
+      _beforeFirstImport ??= node.span.start;
+      _afterLastImport = extendThroughLine(node.span).end;
```

Comment:
What happens here if the user's entire stylesheet is on one line?

Reply:
Good catch. Modified this to add proper logic for finding the safe place to insert additional rules (ideally after a line break, but will leave this insertion point in the middle of a line if the file ends or if there's additional text that's not a comment or whitespace on the same line).